src/common/transformations/src/transformations/common_optimizations/reduce_merge.cpp

```diff
@@ -0,0 +1,104 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/reduce_merge.hpp"
+
+#include <memory>
+#include <ngraph/opsets/opset9.hpp>
+#include <ngraph/pattern/op/or.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/rt_info.hpp>
+
+#include "itt.hpp"
+
+using namespace ngraph;
+
+template <typename T>
+std::shared_ptr<Node> create_pattern() {
+    auto input = pattern::any_input();
+    auto first_axis = pattern::any_input();
+    auto reduce = pattern::wrap_type<T>({input, first_axis});
+    auto second_axis = pattern::any_input();
+    return pattern::wrap_type<T>({reduce, second_axis});
+}
+
+template <typename T>
+bool fuse_reduce_operations(const std::shared_ptr<Node>& node) {
+    const auto bottom_reduce = std::dynamic_pointer_cast<T>(node);
+    if (!bottom_reduce) {
+        return false;
+    }
+
+    const auto top_reduce = std::dynamic_pointer_cast<T>(bottom_reduce->get_input_node_shared_ptr(0));
+    if (!top_reduce) {
+        return false;
+    }
+
+    if (top_reduce->get_keep_dims() != bottom_reduce->get_keep_dims()) {
+        return false;
+    }
+
+    if (!top_reduce->get_keep_dims()) {
+        const auto first_axes = top_reduce->get_reduction_axes();
+        const auto second_axes = bottom_reduce->get_reduction_axes();
+
+        // check if each axis in the second set is smaller than every axis in the first set
+        if (!std::all_of(first_axes.begin(), first_axes.end(), [&second_axes](const size_t first_axis) {
+                return std::all_of(second_axes.begin(), second_axes.end(), [first_axis](const size_t second_axis) {
+                    return first_axis > second_axis;
+                });
+            })) {
+            return false;
+        }
+    }
+
+    auto concat_axes =
+        std::make_shared<opset9::Concat>(OutputVector{top_reduce->input_value(1), bottom_reduce->input_value(1)},
+                                         int64_t(0));
```

Comment:
please call get_constant_from_source on concat_axes to constantfold the axes

Reply:
Fixed